src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2519,33 +2523,20 @@ protected void VisitBinaryInterpolatedStringAddition(BoundBinaryOperator node)
             bool hasConditionalEvaluation = data.UsesBoolReturns || hasTrailingHandlerValidityParameter;
             TLocalState? shortCircuitState = hasConditionalEvaluation ? State.Clone() : default;
 
-            while (stack.TryPop(out var currentString))
+            foreach (var part in parts)
             {
-                visitedFirst |= VisitInterpolatedStringHandlerParts(currentString, data.UsesBoolReturns, firstPartIsConditional: visitedFirst || hasTrailingHandlerValidityParameter, ref shortCircuitState);
+                visitedFirst |= VisitInterpolatedStringHandlerParts(part, data.UsesBoolReturns, firstPartIsConditional: visitedFirst || hasTrailingHandlerValidityParameter, ref shortCircuitState);
             }
 
             if (hasConditionalEvaluation)
             {
                 Join(ref State, ref shortCircuitState);
             }
 
-            stack.Free();
+            parts.Free();
         }
 
-        protected virtual BoundBinaryOperator? PushBinaryOperatorInterpolatedStringChildren(BoundBinaryOperator node, ArrayBuilder<BoundInterpolatedString> stack)
-        {
-            stack.Push((BoundInterpolatedString)node.Right);
-            switch (node.Left)
-            {
-                case BoundBinaryOperator next:
-                    return next;
-                case BoundInterpolatedString @string:
-                    stack.Push(@string);
-                    return null;
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(node.Left.Kind);
-            }
-        }
+        protected virtual void OnPushBinaryOperatorInterpolatedStringChildren(BoundBinaryOperator node) { }
```

Comment:
Could we assert that we're dealing with an additive interpolated string expression?

Reply:
Not easily from this location. There's no flag that indicates that at all levels of the tree, only the top binary operation that has the InterpolatedStringHandlerData could assert this.